libzfs.pyx

```diff
@@ -2892,7 +2892,7 @@ cdef class ZFSPool(object):
     IF HAVE_LZC_SYNC:
         def sync(self, force=False):
             cdef int ret
-            cdef const char *c_name = self.name
+            cdef const char *c_name = libzfs.zpool_get_name(self.handle)
```

Comment:
I don't follow, it's doing the same thing in `name` property ?

Do you have a reproduction case which I could perhaps try locally ?

Reply:
Here's a short script that frequently fails.  Though if I change the pool name to the much smaller and less busy `zroot`, then it passes.
```python
import libzfs
zfs = libzfs.ZFS()
ds = zfs.get_dataset("atl-rb22b-ss.b.24")
pool = ds.pool
pool.sync()
```

Reply:
@asomers can you please create a ticket at https://ixsystems.atlassian.net and link this script there and we can investigate as to why this could be failing potentially. Do mention your scale version please as well. Thanks!

Reply:
What is a "scale version"?

Reply:
BTW I'm not a TrueNAS user.  I'm using a custom Python application on FreeBSD stable/13.

Reply:
@asomers interesting. I am not able to reproduce using the same script ( i am executing the same code path in a loop ). I wonder if it's a cython/python version issue ?
Could you share some more details wrt cython/python version you have please ? Thanks!

Reply:
```
Python 3.9.13
Cython version 0.29.30
```